src/lib.rs

```diff
@@ -172,9 +172,13 @@ where
         A: Clone,
     {
         let q = T::query(query_args.clone());
-        let c: Option<(T, DateTime<Utc>)> = self.cache.get(&q);
+        let initial_state: Option<CacheResult<T>> = self.cache.get(&q);
 
-        self.store.aggregate(query_args, c)
+        self.store.aggregate(query_args, initial_state).map(|agg| {
+            self.cache.insert(&q, agg);
```

Comment:
It would be good to check if it's really required to override the cached values. The new aggregated entity may as well be the same as the already cached value.